pkg/service/service.go

```diff
@@ -24,41 +24,40 @@ const dockerConfigDir = "/var/lib/faasd/.docker/"
 // Remove removes a container
 func Remove(ctx context.Context, client *containerd.Client, name string) error {
 
-	container, containerErr := client.LoadContainer(ctx, name)
-
-	if containerErr == nil {
-		found := true
-		t, err := container.Task(ctx, nil)
-		if err != nil {
-			if errdefs.IsNotFound(err) {
-				found = false
-			} else {
-				return fmt.Errorf("unable to get task %s: ", err)
-			}
+	container, err := client.LoadContainer(ctx, name)
+	if err != nil {
+		service := client.SnapshotService("")
+		key := name + "snapshot"
+		if _, err := client.SnapshotService("").Stat(ctx, key); err == nil {
+			service.Remove(ctx, key)
 		}
+	}
 
-		if found {
-			status, _ := t.Status(ctx)
-			fmt.Printf("Status of %s is: %s\n", name, status.Status)
-
-			log.Printf("Need to kill %s\n", name)
-			err := killTask(ctx, t)
-			if err != nil {
-				return fmt.Errorf("error killing task %s, %s, %s", container.ID(), name, err)
-			}
+	found := true
+	t, err := container.Task(ctx, nil)
+	if err != nil {
+		if errdefs.IsNotFound(err) {
+			found = false
+		} else {
+			return fmt.Errorf("unable to get task %s: ", err)
 		}
+	}
 
-		err = container.Delete(ctx, containerd.WithSnapshotCleanup)
+	if found {
+		status, _ := t.Status(ctx)
+		fmt.Printf("Status of %s is: %s\n", name, status.Status)
+
+		log.Printf("Need to kill %s\n", name)
+		err := killTask(ctx, t)
 		if err != nil {
-			return fmt.Errorf("error deleting container %s, %s, %s", container.ID(), name, err)
-		}
-	} else {
-		service := client.SnapshotService("")
-		key := name + "snapshot"
-		if _, err := client.SnapshotService("").Stat(ctx, key); err == nil {
-			service.Remove(ctx, key)
+			return fmt.Errorf("error killing task %s, %s, %s", container.ID(), name, err)
 		}
 	}
+
+	if err := container.Delete(ctx, containerd.WithSnapshotCleanup); err != nil {
+		return fmt.Errorf("error deleting container %s, %s, %s", container.ID(), name, err)
```

Comment:
Which Go version are we using, in 1.13+ you might want to consider the `%w` verb, it will wrap the error so that the caller can inspect the original error as well.
```suggestion
		return fmt.Errorf("error deleting container %s, %s, %w", container.ID(), name, err)
```

https://blog.golang.org/go1.13-errors#TOC_3.3.

Reply:
Thanks for the tip. I think that's worth considering as a refactoring task.